contracts/truefi2/SAFU.sol

```diff
@@ -27,10 +45,30 @@ contract SAFU is UpgradeableClaimable {
     function liquidate(ILoanToken2 loan) external {
         require(loanFactory.isLoanToken(address(loan)), "SAFU: Unknown loan");
         require(loan.status() == ILoanToken2.Status.Defaulted, "SAFU: Loan is not defaulted");
-        liquidator.liquidate(loan);
+
         ITrueFiPool2 pool = ITrueFiPool2(loan.pool());
+        IERC20 token = IERC20(pool.token());
+
+        liquidator.liquidate(loan);
         pool.liquidate(loan);
-        uint256 lostByPool = loan.debt().mul(loan.balanceOf(address(this))).div(loan.totalSupply());
-        IERC20(pool.token()).safeTransfer(address(pool), lostByPool);
+        uint256 owedToPool = loan.debt().mul(loan.balanceOf(address(this))).div(loan.totalSupply());
+        uint256 safuTokenBalance = tokenBalance(token);
+
+        uint256 toTransfer;
+        uint256 deficit;
+        if (owedToPool <= safuTokenBalance) {
+            deficit = 0;
+            toTransfer = owedToPool;
+        } else {
+            deficit = owedToPool.sub(safuTokenBalance);
+            toTransfer = safuTokenBalance;
+            loanDeficit[loan] = deficit;
+        }
```

Comment:
I personally prefer avoid branching by doing. 
```
uint256 toTransfer = owedToPool;       
uint256 deficit = 0;
if (owedToPool > safuTokenBalance) {
  deficit = owedToPool.sub(safuTokenBalance);            
  toTransfer = safuTokenBalance;            
  loanDeficit[loan] = deficit;
}
``` 
I find it more readable, but it's up for discussion